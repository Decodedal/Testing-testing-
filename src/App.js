import logo from './logo.svg';
import './App.css';
import GithubCard from './GitHubCard';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <h2>Dallas is a code wizard</h2>
        <GithubCard/>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
