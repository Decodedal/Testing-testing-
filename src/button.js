import { Button } from "bootstrap";
import {React, useState} from "react";

function ColorButton(){

    const [Color, changeColor] = useState('red')
   const colorSwitch = Color === "red" ? 'green' : 'red';

    return(
        <button style={{backgroundColor: colorSwitch}} onClick = {()=> changeColor(colorSwitch)} >
            the color is {colorSwitch}
        </button>
    )
}

export default ColorButton;
