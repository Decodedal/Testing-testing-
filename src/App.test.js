import { render, screen } from '@testing-library/react';
import App from './App';
import GithubCard from './GitHubCard';
import button from './button'

import renderer from 'react-test-renderer'
import ColorButton from './button';

// test('renders a snapshot',()=>{
//   const tree = renderer.create(<App/>).toJSON()
//   expect(tree).toMatchSnapshot()
// })

test('renders a snap shot of GitHubCard',()=>{
  const hub = renderer.create(<GithubCard/>).toJSON()
  expect(hub).toMatchSnapshot
})

test('renders a snap shot of button',()=>{
  const butt = renderer.create(<ColorButton/>).toJSON()
  expect(butt).toMatchSnapshot()
})