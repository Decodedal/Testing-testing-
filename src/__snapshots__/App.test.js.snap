// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snap shot of button 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  the color is 
  green
</button>
`;

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
